src/dev/ci_cycles.md

```diff
@@ -23,43 +23,23 @@ the new `ykpack`.
 
 ## How do we Break the Cycle?
 
-1. **Raise PRs**
-
 The change author raises two PRs: one for `yk` and one for `ykrustc`.
 
-2. **yk review**
-
-The reviewer conducts a review of the changes to the `yk` repo first. We get
-the branch squashed and ready, but we don't merge it just yet. If we were to
-try it would fail. Remember: we need the compiler change in first.
-
-3. **Review ykrustc**
-
-Next the reviewer switches to the `ykrustc` pull request and conducts a review
-there. We get this branch squashed and ready to merge, but stop short of
-invoking bors.
-
-4. **Prime the CI**
-
-Next the reviewer should ask the PR author to "prime the CI". A commit should
-be pushed to the ykrustc PR which overrides the yk dependency to the branch
-reviewed in stage 2 of this document. This dependency can be overridden in the
-top-level `Cargo.toml`.
-
-The PR author then issues a `bors try` on the PR, thus caching the new compiler
-in the CI.
+In the `ykrustc` PR description, a line like this should be present:
```

Comment:
I think we can strengthen this a bit. Something like:

> The `ykrustc` PR description must have a line in the format `ci-yk <github_username> <branch>` e.g. `ci-yk vext01 newbranch`. As with `bors` commands, there must be no other text on the line containing this. This line tells the CI to use `github_username:branch` for the `yk` dependency.

Writing that makes me think that we could bikeshed the `ci-yk` bit to something like `ykpr vext01:newbranch`, making the syntax a bit more git-y?

I think we then need to say something like:

> bors will automatically merge both the `ykrustc` and `yk` PRs at the same time.

?

Reply:
We could...

But do we give the false impression that the parameter is passed to git, and that stuff like `:branch` (omitting the upstream) should also work?

Reply:
Point taken.

I do think though (again I'm bikeshedding!) that we might want to kill the `-` since it remind me of "bors-". `ykpr` or `ykbranch` might work?

I also realised that the "PRs will both be merged bit" might be a total hallucination on my part. Any thoughts?

Reply:
TBH, the `-` doesn't bother me too much.

`ykbranch` feels wrong, because it's not just a branch that we are overriding.

`ykpr` is ok, but i prefer `ci-yk` because it can be generalised (thought I hope we don't have to) for other deps too, e.g. `ci-othercrate: ...`.

Reply:
> bors will automatically merge both the ykrustc and yk PRs at the same time.

Yeah, I don't think we can do that without changing bors, I'm afraid.